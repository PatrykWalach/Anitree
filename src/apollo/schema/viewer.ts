export interface User {
  id: number
  name: string
  avata: UserAvatar
  siteUrl: string
  options: {
    profileColor: string
    displayAdultContent: boolean
  }
  mediaListOptions: {
    scoreFormat: ScoreFormat
    animeList: MediaListTypeOptions
    mangaList: MediaListTypeOptions
  }
}
export interface UserAvatar {
  large: string
  medium: string
}
export type ScoreFormat =
  | 'POINT_100'
  | 'POINT_10_DECIMAL'
  | 'POINT_10'
  | 'POINT_5'
  | 'POINT_3'

export interface MediaListTypeOptions {
  customLists: string[]
  advancedScoring: string[]
  advancedScoringEnabled: boolean
}
