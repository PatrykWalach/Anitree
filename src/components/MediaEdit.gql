#import './MediaCardBanner.gql'
#import './MediaCardItem.gql'
#import './MediaEditActions.gql'
#import './MediaEditItems.gql'

fragment MediaEdit_media on Media {
  id
  bannerImage
  ...MediaEditItems_media
  ...MediaEditActions_media
  ...MediaCardBanner_media
  ...MediaCardItem_media
}

fragment MediaEdit_viewer on User {
  id
  ...MediaEditItems_viewer
}
