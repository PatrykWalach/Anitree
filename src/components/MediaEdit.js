import gql from 'graphql-tag'
import { MediaCardBanner_media } from './MediaCardBanner.js'
import { MediaItem_media } from './MediaItem.js'
import { MediaEditActions_media } from './MediaEditActions.js'
import {
  MediaEditItems_media,
  MediaEditItems_viewer,
} from './MediaEditItems.js'

export const MediaEdit_media = gql`
  fragment MediaEdit_media on Media {
    id
    bannerImage
    ...MediaEditItems_media
    ...MediaEditActions_media
    ...MediaCardBanner_media
    ...MediaItem_media
  }
  ${MediaCardBanner_media}
  ${MediaItem_media}
  ${MediaEditActions_media}
  ${MediaEditItems_media}
`

export const MediaEdit_viewer = gql`
  fragment MediaEdit_viewer on User {
    id
    ...MediaEditItems_viewer
  }
  ${MediaEditItems_viewer}
`
