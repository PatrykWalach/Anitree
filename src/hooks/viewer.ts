import { inject } from '@vue/composition-api'
import { useQuery } from '@vue/apollo-composable'
export const DefaultViewer = Symbol('Viewer')

export const useViewer = <V = ReturnType<typeof useQuery>>() =>
  inject(DefaultViewer) as V
