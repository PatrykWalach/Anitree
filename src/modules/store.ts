import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
