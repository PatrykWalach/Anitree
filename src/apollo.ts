import { ApolloLink, concat } from 'apollo-link'
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory'
import { PersistedData, PersistentStorage } from 'apollo-cache-persist/types'

import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'

import { Variables } from '@/graphql/schema/media'
import Vue from 'vue'
import VueApollo from 'vue-apollo'
import { persistCache } from 'apollo-cache-persist'

import { store } from '@/store'

const link = new HttpLink({
  uri: 'https://graphql.anilist.co',
})

const middle = new ApolloLink((operation, forward) => {
  const token = store.getState().settings.token

  if (token) {
    operation.setContext({
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
  }
  if (forward) {
    return forward(operation)
  }
  return null
})

const cache = new InMemoryCache({
  cacheRedirects: {
    Query: {
      Media: (_, args, { getCacheKey }) =>
        getCacheKey({ __typename: 'Media', id: (args as Variables).id }),
    },
  },
})

// if (store.getState()settings..cacheApollo) {
persistCache({
  cache,
  storage: window.localStorage as PersistentStorage<
    PersistedData<NormalizedCacheObject>
  >,
})
// }

export const apollo = new ApolloClient({
  cache,
  link: concat(middle, link),
})

Vue.use(VueApollo)

export const apolloProvider = new VueApollo({
  defaultClient: apollo,
})
