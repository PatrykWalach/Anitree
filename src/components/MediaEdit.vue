<template>
  <v-dialog
    v-model="isEdited"
    scrollable
    :fullscreen="$vuetify.breakpoint.xsOnly"
    max-width="720px"
  >
    <MediaEditLoadingDialog v-if="queryLoading" />
    <MediaEditDialog
      v-else-if="media && viewer"
      :viewer="viewer"
      :media="media"
    />
  </v-dialog>
</template>
<script lang="ts">
import { computed, createComponent } from '@vue/composition-api'
import { useDispatch, useSelector } from 'vue-redux-hooks'
import { useMedia, useViewer } from '@/graphql'
import { useQueryLoading, useResult } from '@vue/apollo-composable'
import MediaEditLoadingDialog from './MediaEditLoadingDialog.vue'
import { State } from '@/store'
import { User } from '../graphql/schema/viewer'
import { asyncComponent } from '@/views/Search.vue'
import { useActions } from '../store'

const MediaEditDialog = () =>
  asyncComponent(
    import(/* webpackChunkName: "MediaEditDialog" */ './MediaEditDialog.vue'),
    MediaEditLoadingDialog,
  )

export interface Props {
  id: number | null
  viewer: User | null
}

export default createComponent<Readonly<Props>>({
  components: {
    MediaEditDialog,
    MediaEditLoadingDialog,
  },
  props: {
    id: { default: null, required: true, type: null },
    viewer: { default: null, required: true, type: null },
  },
  setup(props) {
    const dispatch = useDispatch()

    const {
      edit: { CHANGE_IS_EDITED },
    } = useActions()

    const isEditedState = useSelector((state: State) => state.edit.isEdited)

    const isEdited = computed({
      get: () => isEditedState.value,
      set: isEdited => {
        if (!isEdited) {
          close()
        } else {
          dispatch(CHANGE_IS_EDITED(isEdited))
        }
      },
    })

    const { result } = useViewer()

    const format = useResult(
      result,
      undefined,
      data => data.Viewer.mediaListOptions.scoreFormat,
    )

    const variables = computed(() => ({
      format: format.value,
      id: props.id || 0,
    }))

    const queryLoading = useQueryLoading()

    const { Media: media } = useMedia(variables)

    return {
      isEdited,
      media,
      queryLoading,
    }
  },
})
</script>
