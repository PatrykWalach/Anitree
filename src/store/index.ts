import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
