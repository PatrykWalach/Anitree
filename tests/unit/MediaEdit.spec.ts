import { createVariations, matchSnapshot } from './utils'
import MediaEdit from '@/components/MediaEdit.vue'
import { mockedViewer } from './mocks/viewer'
import { useMockedApollo } from './mocks/apollo'
import { useMockedStore } from './mocks/store'

describe('MediaEdit', () => {
  createVariations({
    propsData: {
      id: () => [null, 1],
      viewer: () => [null, mockedViewer],
    },
  }).forEach(settings =>
    matchSnapshot(MediaEdit, {
      ...settings,
      provide: { ...useMockedStore(), ...useMockedApollo() },
    }),
  )
})
