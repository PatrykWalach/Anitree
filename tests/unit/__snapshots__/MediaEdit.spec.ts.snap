// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaEdit matches snapshot 1`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  maxwidth="720px"
  opendelay="0"
  origin="center center"
  retainfocus="true"
  scrollable="true"
  transition="dialog-transition"
  width="auto"
>
  <!---->
</v-dialog-stub>
`;

exports[`MediaEdit matches snapshot 2`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  maxwidth="720px"
  opendelay="0"
  origin="center center"
  retainfocus="true"
  scrollable="true"
  transition="dialog-transition"
  width="auto"
>
  <!---->
</v-dialog-stub>
`;

exports[`MediaEdit matches snapshot 3`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  maxwidth="720px"
  opendelay="0"
  origin="center center"
  retainfocus="true"
  scrollable="true"
  transition="dialog-transition"
  width="auto"
>
  <!---->
</v-dialog-stub>
`;

exports[`MediaEdit matches snapshot 4`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  maxwidth="720px"
  opendelay="0"
  origin="center center"
  retainfocus="true"
  scrollable="true"
  transition="dialog-transition"
  width="auto"
>
  <!---->
</v-dialog-stub>
`;
